lib/Default/MySqlDatabase.class.php

```diff
@@ -9,16 +9,29 @@ abstract class MySqlDatabase {
 	protected static $selectedDbName;
 	protected $dbResult = null;
 	protected $dbRecord = null;
-	
+
 	public function __construct($dbName) {
 		if (!self::$dbConn) {
 			// Set the mysqli driver to raise exceptions on errors
 			if (!mysqli_report(MYSQLI_REPORT_ERROR | MYSQLI_REPORT_STRICT)) {
 				$this->error('Failed to enable mysqli error reporting');
 			}
 
-			self::$dbConn = new mysqli(self::$host, self::$user, self::$password,
-			                           $dbName, self::$port, self::$socket);
+            $host = self::$host;
+            $user = self::$host;
+            $password = self::$password;
+            $port = self::$port;
+
+			// The configuration can be overridden via PHPUnit configuration
+			if (defined("OVERRIDE_MYSQL_CONFIG")) {
+				$host = constant("OVERRIDE_MYSQL_HOST");
```

Comment:
For these 4 variables, I don't think we need to use the `constant` function. We can just use, e.g.
```php
$host = OVERRIDE_MYSQL_HOST;
```
Or is there a good reason to do it like this? If so, we should do it globally. (The only reason I'm aware of to use this is if the constant name is variable.)

Reply:
The constant names are variables driven by the `phpunit.xml` file, so they have to be retrieved this way. You can do it with raw constants, but if you look in the code, there will be warnings that they are unknown constants.